### PR TITLE
AM2R: Add progressives

### PR DESCRIPTION
**Pull request by Miepee**
Adds progressive suit and jumps
None of them are enabled by default at the moment, as I'd like to only do that once I get non-placeholder sprites for them.

### CODE REVIEW

File: randovania/games/am2r/pickup_database/progressive_items.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from __future__ import annotations
+
+
+def tuples() -> tuple:
```
Comment (ThanatosGit) [inline randovania/games/am2r/pickup_database/progressive_items.py:4]: This should be a list of tuples, where the first element in the list is a string (the progressive name) and the second element is a tuple with an arbitrary amount of strings for the non-progressive name.
So `list[tuple[str, tuple[str, ...]]]` ?

**Review by ThanatosGit — CHANGES_REQUESTED** [review]

File: randovania/games/am2r/pickup_database/progressive_items.py (hunk 1 above)
Comment (henriquegemignani) [reply randovania/games/am2r/pickup_database/progressive_items.py:4]: Would be great to define a type alias somewhere for this mess haha

Comment (Miepee) [reply randovania/games/am2r/pickup_database/progressive_items.py:4]: Yeah, as all the other games will need the same type hint. Should i do that in this PR, or just add a TODO for the future?

Comment (ThanatosGit) [reply randovania/games/am2r/pickup_database/progressive_items.py:4]: Hmmm in feature freeze I wouldn't change anything in other games but it's also just a type hint.
I would say: Add a type alias somewhere where other games can use it in the future, which means don't just put in the AM2R specific file (I hope that there exists a nice spot for it)

File: randovania/games/game.py [hunk 2]
```diff
@@ -32,6 +32,9 @@
     from randovania.layout.preset_describer import GamePresetDescriber
     from randovania.resolver.bootstrap import Bootstrap
 
+# Type alias for progressive items
+Progressive_item_tuples = list[tuple[str, tuple[str, ...]]]
```
Comment (ThanatosGit) [inline randovania/games/game.py:36]: Iirc we use PascalCase so far for a type alias. Like `TeleporterConnection` etc.

**Review by ThanatosGit — CHANGES_REQUESTED** [review]

File: randovania/games/game.py (hunk 2 above)
Comment (henriquegemignani) [reply randovania/games/game.py:36]: Technically it should be `Iterable[tuple[str, tuple[str, ...]]]`, and please also update GameGui.progressive_item_gui_tuples :D

**Review by ThanatosGit — APPROVED** [review]

**Pull request merged**